coveo-settings/coveo_settings/setting_abc.py

```diff
@@ -130,7 +130,7 @@ def key(self) -> str:
     @property
     def value(self) -> Optional[T]:
         """Returns the validated value of the setting, or None when not set."""
-        value = self._get_value()
+        value = settings_adapter.evaluate(self._get_value_before_redirections())
         return None if value is None else self._cast_and_validate(value)

```

Comment:
This is the fix. `get_value()` returns the pre-redirection string (such as env->this-one) and then this is given to `cast_and_validate` which return `-> T` and not `-> Optional[T]`. If it receives None, cast will fail when trying to convert to T.

The solution is to compute the redirection and then only cast/validate if we received a value.